docs/proposals/organizations.md

```diff
@@ -0,0 +1,125 @@
+
+The Organization has a list of members.  Only users from this list may be assigned to OrgGroups.  This prevents an overzealous organization admin from creating OrgGroups made up of non-members.
+
+Organizations will have an `organizations/members` subresource that flattens the list of members by resolving the constituent Users of the MemberGroups.
+
+
+## Quota allocation
+An Organization can be assigned quota by a cluster-admin.  The simplest way to model this is to have a cluster-scoped `ResourceQuota` (We can't call it that because `ResourceQuota` is a namespace-scope resource and it can't be in both scope.) that is linked to an Organization.  The existing quota admission controller would remain in place, but no limits would be set by default.  If an Organization admin decided that a particular project was starving the rest, he could assign hard limits to that Project's `ResourceQuota` to prevent it from taking more.  An additional admission controller would be responsible for doing the rollup across all projects in the Organization and preventing over-use of resources.
+
+This approach is relatively easy to build, but it is limited because it doesn't reserve quota for a given project.  A more complicated approach would be to allow killing of "extra" resources and then provide a rate and ceiling like tc.  This would have an overall cap and allow "borrowing" of quota when resources are plentiful, but then selectively kill resources when resources are scarce to allow for project guarantees.  With enough effort, this sort of solution could be layered on top of kube by setting the ceiling for the existing quota controller and having a separate resource for the guarantee and a separate controller for the killer.
+
+I'm sure there will be complaints no matter what we do.
+
+
+## Default Policy
+org-admins should have power somewhere between a cluster-admin and a project-admin.  Since the idea of an Organization doesn't scope cleanly, we can make use of the `attributeRestrictions` to run custom logic for a cluster-scoped rule.  We can create a `ClusterRoleBinding` to `system:authenticated` that allows a `ClusterRole` with the correct access to every project that the is owned by a Organization that the user is an org-admin for.  That makes it impossible for an org-admin to accidentally (or intentionally) remove his powers on a project.
+
+
+## Project ownership
+Projects will have an annotation that indicates which Organization owns them.  If we don't allow modifications to project by default, then a cluster-admin can sort it out for now until we make an offer/request/approval flow.
+
+
+## Org Member powers
+A given user can:
+ 1.  See which orgs he's a part of.  This will require a reverse-lookup map/cache and probably a subresource on user.
+ 2.  Leave an organization that he's tied to by MemberUser, but not one that he is tied to by MemberGroup.  That's an odd wrinkle.
```

Comment:
> 3) Request to become a member of an(other) org.

I'm hoping to avoid addressing that flow inside this proposal, so I didn't want to list it as a power he'd have.  I'd like to leave that flow out of band for the initial implementation.